Tasks/AzureWebAppV1/deploymentProvider/AzurePublishProfileDeploymentProvider.ts

```diff
@@ -0,0 +1,110 @@
+import tl = require('azure-pipelines-task-lib/task');
+import { KuduServiceUtility } from 'azurermdeploycommon/operations/KuduServiceUtility';
+import { Kudu } from 'azurermdeploycommon/azure-arm-rest/azure-arm-app-service-kudu';
+import { PackageType } from 'azurermdeploycommon/webdeployment-common/packageUtility';
+import { TaskParameters } from '../taskparameters';
+import { IWebAppDeploymentProvider } from './IWebAppDeploymentProvider';
+var zipUtility = require('azurermdeploycommon/webdeployment-common/ziputility.js');
+var deployUtility = require('azurermdeploycommon/webdeployment-common/utility.js');
+
+var parseString = require('xml2js').parseString;
+
+interface scmCredentials {
+    uri: string;
+    username: string;
+    password: string;
+}
+
+export class AzurePublishProfileDeploymentProvider implements IWebAppDeploymentProvider {
```

Comment:
Do we need a separate DeploymentProvider? Or do we just need to extend creating kuduService ?

Reply:
if there are sub-functionalities which are not applicable, we should handle just those. for example, I believe web package transformation is not possible with publish profile. we should just use package path directly.

Reply:
Another angle I am thinking is that in next version of task when all such incompatible functionalities are removed, this deployment provider would become redundant and duplicate code.